test-resources-testcontainers/src/main/java/io/micronaut/testresources/testcontainers/TestContainerMetadataSupport.java

```diff
@@ -225,6 +238,82 @@ public String getValue() {
             .orElse(Collections.emptyMap());
     }
 
+    private static WaitStrategy extractWaitStrategyFrom(String prefix, Map<String, Object> testResourcesConfiguration) {
```

Comment:
Why don't you use a configuration class and let Micronaut parse and validate it for you?

Reply:
There is no bean context

Reply:
Correct: test resources are setup _before_ the bean context is available.